### PR TITLE
Add internal pixelmap count to annotation/group count

### LINKED ISSUE #200: Add group count for the cells within pixelmaps
As mentioned in https://github.com/DigitalSlideArchive/HistomicsUI/issues/198#issuecomment-1177796363 it would be nice to add a group count for the cells of individual pixelmaps similar to the count added for separate annotation elements.

### PR DESCRIPTION
**Pull request by eagw**
Counts the number of cells within a pixelmap that belong to each annotation group and adds the totals to the existing count for individual elements.  Also updates the count when changes are made to the groups of the cells within the pixelmap or when the pixelmap is deleted.

Closes #200

### CODE REVIEW

**Comment by manthey** [comment]
What determines the order the groups are listed in?  We should probably have an obvious order, like most numerous to least numerous and alphabetical when counts are identical.  But if there is already a rationale for the order, we don't have to change.

**Comment by manthey** [comment]
When we hold down shift and drag to mark categories, the counts only update on mouse up.  If it isn't too expensive, it would be nice if it updated as you drag.

**Comment by eagw** [comment]
> When we hold down shift and drag to mark categories, the counts only update on mouse up. If it isn't too expensive, it would be nice if it updated as you drag.

As of now the count updates when the draw widget is re-rendered after the mouse is let up (or enough time passes without the mouse moving) and as far as I can tell the only way to update more frequently than this would be to update the count every time the overlay mouseover event is called.  Since this happens so frequently it results in the numbers being counted too many times, so I think it's best to stay with the frequency it currently has.

**Comment by manthey** [comment]
> > When we hold down shift and drag to mark categories, the counts only update on mouse up. If it isn't too expensive, it would be nice if it updated as you drag.
> 
> As of now the count updates when the draw widget is re-rendered after the mouse is let up (or enough time passes without the mouse moving) and as far as I can tell the only way to update more frequently than this would be to update the count every time the overlay mouseover event is called. Since this happens so frequently it results in the numbers being counted too many times, so I think it's best to stay with the frequency it currently has.

Is updating as part of the `_updatePixelmapValues` feasible (or `_redrawAnnotation`)?  The `_updatePixelmapValues` function probably should have a guard that if the data hasn't changed, we shouldn't call `pixelmapElementModel.set('values', newData);` or `_redrawAnnotation`.

**Comment by eagw** [comment]
> Is updating as part of the `_updatePixelmapValues` feasible (or `_redrawAnnotation`)? The `_updatePixelmapValues` function probably should have a guard that if the data hasn't changed, we shouldn't call `pixelmapElementModel.set('values', newData);` or `_redrawAnnotation`.

It's feasible, but based on my testing `_updatePixelmapValues` is actually called with the same frequency that the draw widget is re-rendered (ie. when you let go of the mouse) and `_redrawAnnotation` is called even less often, so unfortunately it wouldn't result in the count updating more frequently.

File: histomicsui/web_client/panels/DrawWidget.js [hunk 1]
```diff
@@ -725,6 +770,21 @@ var DrawWidget = Panel.extend({
         }
     },
 
+    countPixelmap(pixelmap, operation) {
+        let toChange = {};
+        for (let ix = 0; ix < pixelmap.attributes.values.length / 2; ix++) {
```
Comment (manthey) [inline histomicsui/web_client/panels/DrawWidget.js:775]: I think the `/ 2` is only required if `pixelmap.get('boundaries')` is truthy.

Comment (eagw) [reply histomicsui/web_client/panels/DrawWidget.js]: Okay- see 5bc64c0.

**Review by manthey — APPROVED** [review]
Thank you.

**Pull request merged**